client/state/selectors/get-magic-login-email-address-form-input-is-valid.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * External dependencies
+ */
+import emailValidator from 'email-validator';
+
+/**
+ * Internal dependencies
+ */
+import getMagicLoginEmailAddressFormInput from './get-magic-login-email-address-form-input';
+
+export default function getMagicLoginEmailAddressFormInputIsValid( state ) {
+	const email = getMagicLoginEmailAddressFormInput( state );
+	return !! (
+		typeof email === 'string' &&
+		emailValidator.validate( email )
+	);
+}
```

Comment:
**Much** better for this to be a selector instead of a reducer with non-essential state. :)